PurchasesCoreSwift/Networking/Backend.swift

```diff
@@ -0,0 +1,650 @@
+public typealias IntroEligibilityResponseHandler = ([String: IntroEligibility], Error?) -> Void
+public typealias OfferingsResponseHandler = ([String: Any]?, Error?) -> Void
+public typealias OfferSigningResponseHandler = (String?, String?, UUID?, NSNumber?, Error?) -> Void
+
+// swiftlint:disable type_body_length file_length
+// TODO(post-migration): Make this internal again, and all the other things too
+@objc(RCBackend) public class Backend: NSObject {
+
+    @objc public static let RCSuccessfullySyncedKey: NSError.UserInfoKey = "rc_successfullySynced"
+    @objc public static let RCAttributeErrorsKey = "attribute_errors"
+    static let RCAttributeErrorsResponseKey = "attributes_error_response"
+
+    private let httpClient: HTTPClient
+    private let apiKey: String
+
+    // callbackQueue controls access to both offeringsCallbacksCache and purchaserInfoCallbacksCache
+    private let callbackQueue = DispatchQueue(label: "Backend callbackQueue")
+    private var offeringsCallbacksCache: [String: [OfferingsResponseHandler]]
+    private var purchaserInfoCallbacksCache: [String: [BackendPurchaserInfoResponseHandler]]
+
+    private var authHeaders: [String: String] { return ["Authorization": "Bearer \(self.apiKey)"] }
+
+    @objc(initWithAPIKey:systemInfo:eTagManager:operationDispatcher:)
+    public convenience init(apiKey: String,
+                            systemInfo: SystemInfo,
+                            eTagManager: ETagManager,
+                            operationDispatcher: OperationDispatcher) {
+        let httpClient = HTTPClient(systemInfo: systemInfo,
+                                    eTagManager: eTagManager,
+                                    operationDispatcher: operationDispatcher)
+        self.init(httpClient: httpClient, apiKey: apiKey)
+    }
+
+    @objc public required init(httpClient: HTTPClient, apiKey: String) {
+        self.httpClient = httpClient
+        self.apiKey = apiKey
+        self.offeringsCallbacksCache = [:]
+        self.purchaserInfoCallbacksCache = [:]
+    }
+
+    @objc(createAliasForAppUserID:newAppUserID:completion:)
+    public func createAlias(appUserID: String, newAppUserID: String, completion: ((Error?) -> Void)?) {
+        guard let appUserID = try? escapedAppUserID(appUserID: appUserID) else {
+            completion?(ErrorUtils.missingAppUserIDError())
+            return
+        }
+
+        let path = "/subscribers/\(appUserID)/alias"
+        httpClient.performPOSTRequest(serially: true,
+                                      path: path,
+                                      requestBody: ["new_app_user_id": newAppUserID],
+                                      headers: authHeaders) { statusCode, response, error in
+            self.handle(response: response, statusCode: statusCode, error: error, completion: completion)
+        }
+    }
+
+    @objc public func clearCaches() {
+        httpClient.clearCaches()
+    }
+
+    // swiftlint:disable function_parameter_count
```

Comment:
[not for this PR] we should probably just start creating structs for these when they get too big. `ProductInfo` was actually meant to help, but there's still a bunch more params

Reply:
💯